shell/server/core.js

```diff
@@ -147,6 +148,64 @@ class NotificationHandle {
   }
 }
 
+class PersistentUiViewImpl {
+  constructor(token, sturdyRef, membraneRequirements) {
+    this._token = token;
+    this._sturdyRef = sturdyRef;
+    // We need this token so we can later pass it to SandstormPermissions.createNewApiToken, which
+    // requires the sturdyRef of the token it's producing a child for, not just the hashed token id.
```

Comment:
I don't think there would be any problem with allowing "createNewApiToken()` to accept a `{hashedParentToken: String}` provider. Indeed, we'll need that ability if we ever implement the "Asheesh button" for generating child tokens from the Who Has Access view.

It's interesting that `rawParentToken` works for the present use case, though!